src/proficiency.cpp

```diff
@@ -658,3 +670,40 @@ float book_proficiency_bonuses::time_factor( const proficiency_id &id ) const
 
     return static_cast<float>( 1.0 - std::exp( -std::sqrt( sum ) ) );
 }
+
+void proficiency_migration::load( const JsonObject &jo )
+{
+    proficiency_migration migration;
+    mandatory( jo, false, "from", migration.id_old );
+    mandatory( jo, false, "to", migration.id_new );
+    prof_migrations.emplace( migration.id_old, migration );
+}
+
+void proficiency_migration::reset()
+{
+    prof_migrations.clear();
+}
+
+void proficiency_migration::check()
+{
+    for( const auto &[from_id, _] : prof_migrations ) {
+        const proficiency_migration &pm = *find_migration( from_id );
+        if( !pm.id_new.is_valid() ) {
+            debugmsg( "proficiency migration specifies invalid id '%s'", pm.id_new.str() );
+            continue;
+        }
+    }
+}
+
+const proficiency_migration *proficiency_migration::find_migration( const proficiency_id &original )
+{
+    proficiency_migration *migration = nullptr;
+
+    const auto &migration_it = prof_migrations.find( original );
+    if( migration_it == prof_migrations.cend() ) {
+        return migration;
+    }
+    migration = &migration_it->second;
+
+    return migration;
```

Comment:
Typically you don't make references (even const ones) to iterators.
You can also just return the values directly, instead of making a variable to store them.
```suggestion
    const auto migration_it = prof_migrations.find( original );
    if( migration_it == prof_migrations.cend() ) {
        return nullptr;
    }
    return &migration_it->second;
```

Reply:
That was a copy (with some adjustments) of vpart_migration::find_migration(), but okay